wave_matching/include/wave/matching/matcher.hpp

```diff
@@ -0,0 +1,51 @@
+#ifndef WAVE_MATCHING_MATCHER_HPP
+#define WAVE_MATCHING_MATCHER_HPP
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+namespace wave {
+
+// sensor data type
+template <typename T>
+class Matcher {
+ public:
```

Comment:
I think we might want to have an exception to that rule. I'm in favor of no indent, unless you have problems with it? 

@michaelhsmart? thoughts?